File: src/main/java/no/novari/flyt/archive/gateway/template/model/Dependency.java
package no.novari.flyt.archive.gateway.template.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.Valid;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;

import java.util.Collection;

@Getter
@Builder
@EqualsAndHashCode
@Jacksonized
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Dependency {
    private final Collection<Collection<@Valid ValuePredicate>> hasAnyCombination;
}

File: src/main/kotlin/no/novari/flyt/archive/gateway/template/model/Dependency.kt
package no.novari.flyt.archive.gateway.template.model

import com.fasterxml.jackson.annotation.JsonInclude
import jakarta.validation.Valid

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Dependency(
    val hasAnyCombination: Collection<Collection<@Valid ValuePredicate>>? = null,
) {
    companion object {
        @JvmStatic
        fun builder() = Builder()
    }

    class Builder {
        private var hasAnyCombination: Collection<Collection<ValuePredicate>>? = null

        fun hasAnyCombination(hasAnyCombination: Collection<Collection<ValuePredicate>>?) =
            apply {
                this.hasAnyCombination = hasAnyCombination
            }

        fun build() = Dependency(hasAnyCombination = hasAnyCombination)
    }
}
